python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -302,3 +316,47 @@ def _generate_ir_tasks_pwise(
 generate_ir_tasks.register(Filter, _generate_ir_tasks_pwise)
 generate_ir_tasks.register(HStack, _generate_ir_tasks_pwise)
 generate_ir_tasks.register(Select, _generate_ir_tasks_pwise)
+generate_ir_tasks.register(Distinct, _generate_ir_tasks_pwise)
+
+
+@lower_ir_node.register(Distinct)
+def _(
+    ir: Distinct, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    """
+    Lower a Distinct node.
+
+    Distinct is implemented as a set of per-partition Distinct operations,
+    followed by a Shuffle, followed by a final Distinct.
+    """
+    child, partition_info = rec(ir.children[0])
+    subset = ir.subset or set(ir.schema)
+    partitioned_on = tuple(
+        NamedExpr(name, Col(dtype, name))
+        for name, dtype in ir.schema.items()
+        if name in subset
+    )
+
+    output_count = partition_info[child].count
+    if output_count == 1:
+        new_node = ir.reconstruct([child])
+        partition_info[new_node] = PartitionInfo(count=1, partitioned_on=partitioned_on)
+        return new_node, partition_info
+    else:
+        # Do a shuffle, by
+        shuffle_options = ConfigOptions({})
```

Comment:
Might be best to call this `config_options`, since it actually holds **all** user-defined options.